compiler/semantic.ml

```diff
@@ -28,33 +42,137 @@ let string_to_data_type (s: string) = match s
 	| "json" -> Json
 	| _ -> raise (Failure "unsupported data type")
 
+let check_binop_type (left_expr : data_type) (op : Ast.math_op) (right_expr : data_type) = match (left_expr, op, right_expr)
+	with (Int, _, Int) -> Int
+	| (Float, _, Float) -> Float
+	| (String, Add, String) -> String
+	| (_, _, _) -> raise (Failure "cannot perform binary operations with provided arguments")
 
-let check_expr_data_type (expr : Ast.expr) = match expr
+let rec check_bracket_select_type (d_type : data_type) (selectors : expr list) (env : symbol_table) (id : string) = match d_type
```

Comment:
How does `(env : symbol_table)` work here? In other places in `semantic.ml`, you specify `Environment.symbol_table`, which seems to make more sense, because `symbol_table` is defined in `environment.ml`